### PR TITLE
New Dataset structure

### PR DESCRIPTION
**Pull request by VasilievaKA**
Added new abstract classes 
Refactored tutorials

### CODE REVIEW

File: hypex/dataset/dataset.py [hunk 1]
```diff
@@ -320,23 +199,40 @@ def groupby(
         return datasets
 
     def mean(self):
-        return Dataset(data=self._backend.mean(), roles=self.roles)
+        return Dataset(
```
Comment (Dmatryus) [inline hypex/dataset/dataset.py:202]: return dataset or float

File: hypex/utils/errors.py [hunk 2]
```diff
@@ -22,15 +22,24 @@ class SpaceError(Exception):
...
 
 
 class NotFoundInExperimentDataError(Exception):
     def __init__(self, class_: type):
-        super().__init__(f"{class_.__name__} id is not found in ExperimentData")
\ No newline at end of file
+        super().__init__(f"{class_.__name__} id is not found in ExperimentData")
+
+
+class AbstractMethodError(Exception):
```
Comment (Dmatryus) [inline hypex/utils/errors.py:41]: May be NotImplementException?

File: hypex/dataset/dataset.py [hunk 3]
```diff
@@ -320,23 +199,40 @@ def groupby(
         return datasets
 
     def mean(self):
```
Comment (Dmatryus) [inline hypex/dataset/dataset.py:201]: Add notation 
`-> Union[Dataset, float]`

File: hypex/utils/errors.py (hunk 2 above)
Comment (VasilievaKA) [reply hypex/utils/errors.py:41]: then what's the difference between NotImplementException and NotImplementedError?

**Review by Dmatryus — APPROVED** [review]
😶‍🌫️

**Pull request merged**